File: CoreProjects/atlassian-packages/rankevolve/src/server/llm/anthropic_client.py
# (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
# pyre-strict

from __future__ import annotations

from collections.abc import AsyncIterator

import anthropic

from rankevolve.src.server.llm.base import BaseLLMClient


class AnthropicClient(BaseLLMClient):
    """Streaming client for the Anthropic Messages API."""

    def __init__(self, api_key: str, base_url: str | None = None) -> None:
        kwargs: dict[str, str] = {"api_key": api_key}
        if base_url:
            kwargs["base_url"] = base_url
        self.client = anthropic.AsyncAnthropic(**kwargs)

    async def stream_response(
        self,
        messages: list[dict[str, str]],
        system: str,
        model: str,
        max_tokens: int,
        temperature: float,
    ) -> AsyncIterator[str]:
        async with self.client.messages.stream(
            model=model,
            max_tokens=max_tokens,
            temperature=temperature,
            system=system,
            messages=messages,  # pyre-ignore[6]
        ) as stream:
            async for text in stream.text_stream:
                yield text

    async def close(self) -> None:
        await self.client.close()

File: CoreProjects/atlassian-packages/rankevolve/src/cli/chat_cli_ts/src/llm/anthropic-client.ts
import Anthropic from "@anthropic-ai/sdk";

import type { LLMClient, StreamParams } from "./base.js";

export class AnthropicClient implements LLMClient {
  private client: Anthropic;

  constructor(apiKey: string, baseUrl?: string) {
    this.client = new Anthropic({
      apiKey,
      ...(baseUrl ? { baseURL: baseUrl } : {}),
    });
  }

  async *streamResponse(params: StreamParams): AsyncIterable<string> {
    const stream = this.client.messages.stream({
      model: params.model,
      max_tokens: params.maxTokens,
      temperature: params.temperature,
      system: params.system,
      messages: params.messages as Anthropic.MessageParam[],
    });

    for await (const event of stream) {
      if (
        event.type === "content_block_delta" &&
        event.delta.type === "text_delta"
      ) {
        yield event.delta.text;
      }
    }
  }

  async close(): Promise<void> {
    // Anthropic TS SDK doesn't require explicit close
  }
}
